metrics-operator/converter/slo_converter.go

```diff
@@ -163,30 +171,54 @@ func setupTarget(o *Objective) (*metricsapi.Target, error) {
 		}
 	}
 
-	// if warning criteria are defined, create new criteria with the following logic:
+	// if pass is superinterval of warn, the following logic is used:
+	// !(pass criteria) -> fail criteria
+	// warn criteria -> warn criteria
+	if isSuperInterval(o.Pass[0].Operators, o.Warning[0].Operators) {
+		op1, err := newOperator(o.Warning[0].Operators, false)
+		if err != nil {
+			return target, err
+		}
+		op2, err := newOperator(o.Pass[0].Operators, true)
+		if err != nil {
+			return target, err
+		}
+		target.Failure = op2
+		target.Warning = op1
+		return target, nil
+	}
+
+	// if warning is superinterval of pass OR we have a single rule criteria, the following logic is used:
 	// !(warn criteria) -> fail criteria
 	// !(pass criteria) -> warn criteria
-	var err error
-	if len(o.Pass) > 0 {
-		if len(o.Pass[0].Operators) > 0 {
-			// TODO cover use cases with multiple operators (create new objectives)
-			op, err := newOperator(o.Pass[0].Operators[0])
-			if err != nil {
-				return target, err
-			}
-			target.Warning = op
+	// TODO change if statement when isSuporInterval is implemented
+	if (len(o.Pass[0].Operators) == 1 && len(o.Warning[0].Operators) == 1) || true /*isSuperInterval(o.Warning[0].Operators, o.Pass[0].Operators) */ {
+		op1, err := newOperator(o.Pass[0].Operators, true)
+		if err != nil {
+			return target, err
 		}
-		if len(o.Warning[0].Operators) > 0 {
-			// TODO cover use cases with multiple operators (create new objectives)
-			op, err := newOperator(o.Warning[0].Operators[0])
-			if err != nil {
-				return target, err
-			}
-			target.Failure = op
+		op2, err := newOperator(o.Warning[0].Operators, true)
+		if err != nil {
+			return target, err
 		}
+		target.Failure = op2
+		target.Warning = op1
+		return target, nil
 	}
+	return target, nil
+}
 
-	return target, err
+// TODO implement
+func isSuperInterval(op1 []string, op2 []string) bool {
+	// superOp1, superOpVal1, err := decodeOperatorAndValue(op1[0])
+	// if err != nil {
+	// 	return false, err
+	// }
+	// subOp1, subOpVal1, err := decodeOperatorAndValue(op2[0])
+	// if err != nil {
+	// 	return false, err
+	// }
```

Comment:
why the logic is commented out?

Reply:
this logic is preparation for future adaptations and follow-up, which is already in progress

Reply:
adapted and removed it for now